demos/interactive_face_detection_demo/cpp/main.cpp

```diff
@@ -71,63 +63,25 @@ int main(int argc, char *argv[]) {
-        AntispoofingClassifier antispoofingClassifier(FLAGS_m_am, FLAGS_d_am, FLAGS_n_am, FLAGS_dyn_am, FLAGS_async, FLAGS_r);
-
-        for (auto && option : cmdOptions) {
-            auto deviceName = option.first;
-            auto networkName = option.second;
-
-            if (deviceName.empty() || networkName.empty()) {
-                continue;
-            }
-
-            if (loadedDevices.find(deviceName) != loadedDevices.end()) {
-                continue;
-            }
-
-            /** Loading extensions for the CPU device **/
-            if ((deviceName.find("CPU") != std::string::npos)) {
+        slog::info << ov::get_openvino_version() << slog::endl;
+        ov::Core core;
 
-                if (!FLAGS_l.empty()) {
-                    // CPU(MKLDNN) extensions are loaded as a shared library and passed as a pointer to base extension
-                    auto extension_ptr = std::make_shared<InferenceEngine::Extension>(FLAGS_l);
-                    ie.AddExtension(extension_ptr, "CPU");
-                }
-            } else if (!FLAGS_c.empty()) {
-                // Loading extensions for GPU
-                ie.SetConfig({{InferenceEngine::PluginConfigParams::KEY_CONFIG_FILE, FLAGS_c}}, "GPU");
-            }
-
-            loadedDevices.insert(deviceName);
-        }
+        FaceDetection faceDetector(FLAGS_m, FLAGS_t, FLAGS_r,
+                                   static_cast<float>(FLAGS_bb_enlarge_coef), static_cast<float>(FLAGS_dx_coef), static_cast<float>(FLAGS_dy_coef));
+        AgeGenderDetection ageGenderDetector(FLAGS_m_ag, FLAGS_r);
+        HeadPoseDetection headPoseDetector(FLAGS_m_hp, FLAGS_r);
+        EmotionsDetection emotionsDetector(FLAGS_m_em, FLAGS_r);
+        FacialLandmarksDetection facialLandmarksDetector(FLAGS_m_lm, FLAGS_r);
+        AntispoofingClassifier antispoofingClassifier(FLAGS_m_am, FLAGS_r);
         // ---------------------------------------------------------------------------------------------------
 
         // --------------------------- 2. Reading IR models and loading them to plugins ----------------------
-        // Disable dynamic batching for face detector as it processes one image at a time
-        Load(faceDetector).into(ie, FLAGS_d, false);
-        Load(ageGenderDetector).into(ie, FLAGS_d_ag, FLAGS_dyn_ag);
-        Load(headPoseDetector).into(ie, FLAGS_d_hp, FLAGS_dyn_hp);
-        Load(emotionsDetector).into(ie, FLAGS_d_em, FLAGS_dyn_em);
-        Load(facialLandmarksDetector).into(ie, FLAGS_d_lm, FLAGS_dyn_lm);
-        Load(antispoofingClassifier).into(ie, FLAGS_d_am, FLAGS_dyn_am);
+        Load(faceDetector).into(core, FLAGS_d);
+        Load(ageGenderDetector).into(core, FLAGS_d);
+        Load(headPoseDetector).into(core, FLAGS_d);
+        Load(emotionsDetector).into(core, FLAGS_d);
+        Load(facialLandmarksDetector).into(core, FLAGS_d);
+        Load(antispoofingClassifier).into(core, FLAGS_d);
```

Comment:
MULTI uses greedy approach when choosing which device to use for current request. But mostly yes, MULTI should be sufficient because people are unlikely to mess with choosing which particular device to assign to each of 6 nets.

`-d MULTI:CPU,CPU` works, but even just `-d GPU` fails with `Only dynamic batch is supported!`. That's broken GPU plugin.